### DISCUSSION TITLE
What changed between 3.0.7 and 3.1 that requires 16 additional nuget package dependencies

### DISCUSSION THREAD

**Discussion by cedral** [quartznet/quartznet#916] [General] [answered]
That's kind of a nightmare for anybody trying to maintain an installer. Is that a mistake of some kind? it doesn't seem like that much changed between the 2 versions.

**Comment by cedral** [comment]
I guess it is mostly Microsoft.Data.SQLClient. What does that gain us as a dependency? It's awfully expensive.

**Comment by lahma** [comment, marked as answer]
This is a known issue and will be fixed as part of version 3.2: https://github.com/quartznet/quartznet/issues/903 . Quartz will no longer have direct binary package dependency on Microsoft.Data.SqlClient.

**Reply by cedral** [reply]
Is this just making the dependency effectively lazy or will it make it agnostic about whether Microsoft.Data.SqlClient or System.Data.SqlClient is used? I ask because I am using the full framework, not core.  With the full framework System.Data.SqlClient is provided as part of the system and I don't have to distribute it. If using SQL server requires Microsoft.Data.SqlClient going forward making it lazy won't help me because I am  using SQL Server.

**Reply by lahma** [reply]
I think we can refine the solution a bit further. 

* Make provider `SqlServer` a "dynamic" provider
* When on full framework this resolves to `SystemDataSqlClient` (we bring this back for full framework, not available on .NET Core)
* When on .NET Core this resolves to `MicrosoftDataSqlClient` (the current one)

So you can generally use `SqlServer` as provider and you get 3.0 behavior on full framework with Quartz 3.2. You can opt to use `MicrosoftDataClient` by using it as provider name (it has more features than System.Data.SqlClient). For .NET Core you need to install the Microsoft.Data.SqlClient dependency, always.

**Comment by lahma** [comment]
Now with https://github.com/quartznet/quartznet/pull/921 System.Data.SqlClient will be again default for full framework, Microsoft.Data.SqlClient can be configured as provider if wanted.
